File: remove_duplicates.rb
def remove_duplicates(nums)
    nums.each_with_index do |n, i|
        if n == nums[i+1]
            nums[i] = ""
        end
    end
    nums = nums.select { |n| n != "" }
    return nums.length
end

puts remove_duplicates([1, 1, 1, 2, 2, 3, 3])

File: javaScript/remove_duplicates.js
var removeDuplicates = function(nums) {
    for (i = 0; i < nums.length - 1; i++) {
        if (nums[i] === nums[i + 1]) {
            nums.splice(i, 1)
            i -= 1
        }    
    }
    return nums.length
};